client/app/bundles/comments/containers/NavigationBarContainer.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import NavigationBar from '../components/NavigationBar/NavigationBar';
+import * as commentsActionCreators from '../actions/commentsActionCreators';
+import BaseComponent from 'libs/components/BaseComponent';
+
+function stateToProps(state) {
+  // Which part of the Redux global state does our component want to receive as props?
+  if (state.$$commentsStore) {
+    return {
+      commentsCount: state.$$commentsStore.get('$$comments').size,
+      pathname: state.railsContext.pathname,
+    };
+  } else {
+    return { };
+  }
+}
+
```

Comment:
@justin808 [reselect](https://github.com/reactjs/reselect). It's not necessary in a trivial example like this, but if you want to exemplify our practices on our internal project, then yes, we definitely use it so that we can decouple our dumb components from the structure of our stores and also convert immutable js objects to native without making huge performance sacrifices that doing this would normally bring (since reselect memoizes)

react-select is the autosuggest thing